examples/nextjs/src/components/layout/header/header.js

```diff
@@ -1,17 +1,30 @@
+'use client';
 import Link from 'next/link';
+import { isInsideEditor } from '@dotcms/client';
+import { useEffect, useMemo, useState } from 'react';
 import ReorderButton from './components/reorderMenu';
 
 function Header({ children }) {
+    const [insideEditor, setInsideEditor] = useState(false);
+
+    useEffect(() => {
+        setInsideEditor(isInsideEditor());
+    }, [])
```

Comment:
you can use a `useMemo`, check contentlets component

Reply:
For some reason, useMemo on this components trigger an Hydration 
`error: Hydration failed because the initial UI does not match what was rendered on the server.`

Copilot say It is preferable use useEffect to client-side updates, so after talk with the team, we will keep here a useEffect here

![Screenshot 2024-11-19 at 2 03 43 PM](https://github.com/user-attachments/assets/e6a9d54c-3552-4cf3-b379-acce8a6affa0)